README.md
# R.archive: Automatic Copying of Objects and Files to a File-based Archive


## Installation
R package R.archive is only available via [GitHub](https://github.com/HenrikBengtsson/R.archive) and can be installed in R as:
```r
source('http://callr.org/install#HenrikBengtsson/R.archive')
```

### Pre-release version

To install the pre-release version that is available in branch `develop`, use:
```r
source('http://callr.org/install#HenrikBengtsson/R.archive@develop')
```
This will install the package from source.  



## Software status

| Resource:     | GitHub        | Travis CI     | Appveyor         |
| ------------- | ------------------- | ------------- | ---------------- |
| _Platforms:_  | _Multiple_          | _Linux_       | _Windows_        |
| R CMD check   |  | <a href="https://travis-ci.org/HenrikBengtsson/R.archive"><img src="https://travis-ci.org/HenrikBengtsson/R.archive.svg" alt="Build status"></a> | <a href="https://ci.appveyor.com/project/HenrikBengtsson/r-archive"><img src="https://ci.appveyor.com/api/projects/status/github/HenrikBengtsson/R.archive?svg=true" alt="Build status"></a> |
| Test coverage |                     | <a href="https://coveralls.io/r/HenrikBengtsson/R.archive"><img src="https://coveralls.io/repos/HenrikBengtsson/R.archive/badge.svg?branch=develop" alt="Coverage Status"/></a>   |                  |


NEWS
Package: R.archive
==================

Version: 0.3.1-9000 [2015-12-27]
o ...


Version: 0.3.1 [2015-12-18]
o Updated package dependencies.

 
Version: 0.3.0 [2013-10-28]
o Updated package dependencies.
o Package requires R (>= 3.0.2).
o ROBUSTNESS: Now importing only what needs to be imported and
  formally declaring all S3 methods in NAMESPACE.
o CLEANUP: Now package attaches no other package when loaded, i.e.
  they're only imported.
o CLEANUP: Dropped obsolete usage of autoload().


Version: 0.2.0 [2012-11-22]
o DOCUMENTATION: Minor clarifications and fixed a broken cross link.
o ROBUSTNESS: Explicitly declaring S3 methods.
o CLEANUP: Now exporting fewer methods.
o CLEANUP: Now only importing needed methods from R.methodsS3 
  and R.oo.  R.utils still needs to be loaded.


Version: 0.1.2 [2011-03-18]
o In order to make it more unique, the timestamp added to files now
  includes milliseconds as well, e.g. 041359.032.


Version: 0.1.1 [2011-03-10]
o Added get- and setArchiveOption().
o Added get- and setArchiveRootPath().


Version: 0.1.0 [2011-03-09]
o Added archiveFile() for adding a file to the archive directory.
  It makes use of internal utility function getArchivePath(), 
  getArchiveFilename(), and getArchivePathname().
o Created.


R/getArchiveRootPath.R
#########################################################################/**
# @RdocDefault getArchiveRootPath
#
# @title "Gets the root path to the file archive directory"
#
# \description{
#  @get "title".
# }
#
# @synopsis
#
# \arguments{
#   \item{defaultPath}{The default path, if no user-specified directory
#     has been given.}
#   \item{...}{Not used.}
# }
#
# \value{
#   Returns the archive root path as a @character string.
# }
#
# @author
#
# \seealso{
#  Too set the directory where archived files are stored,
#  see @see "setArchiveRootPath".
# }
#
# @keyword "programming"
# @keyword "IO"
#*/#########################################################################
setMethodS3("getArchiveRootPath", "default", function(defaultPath="~/.Rarchive", ...) {
  getArchiveOption("rootPath", defaultPath);
})



#########################################################################/**
# @RdocDefault setArchiveRootPath
#
# @title "Sets the root path to the file archive directory"
#
# \description{
#  @get "title".
# }
#
# @synopsis
#
# \arguments{
#   \item{path}{The path.}
#   \item{...}{Not used.}
# }
#
# \value{
#   Returns (invisibly) the old root path.
# }
#
# @author
#
# \seealso{
#  @see "getArchiveRootPath".
# }
#
# @keyword "programming"
# @keyword "IO"
#*/#########################################################################
setMethodS3("setArchiveRootPath", "default", function(path="~/.Rarchive", ...) {
  if (!is.null(path)) {
    path <- as.character(path)
    if (!isDirectory(path)) mkdirs(path, mustWork=TRUE)
  }

  ovalue <- setArchiveOption("rootPath", path)

  invisible(ovalue)
})


############################################################################
# HISTORY:
# 2011-03-10
# o Added get- and setArchiveRootPath().  Adapted from R.cache.
# o Created.
############################################################################


R/getArchivePath.R
#########################################################################/**
# @RdocDefault getArchivePath
#
# @title "Gets the path to the file archive directory"
#
# \description{
#  @get "title".
# }
#
# @synopsis
#
# \arguments{
#   \item{dirs}{A @character @vector constituting the path to the
#      archive subdirectory to be used.  If @NULL, the root path
#      is used.}
#   \item{...}{Not used.}
# }
#
# \value{
#   Returns the archive path as a @character string.
# }
#
# @author
#
# \seealso{
#   @see "getArchiveOption".
#   @see "getArchiveRootPath".
# }
#
# @keyword "programming"
# @keyword "IO"
# @keyword "internal"
#*/#########################################################################
setMethodS3("getArchivePath", "default", function(dirs=NULL, ...) {
  # Argument 'dirs'
  dirs <- Arguments$getCharacters(dirs);

  # Generate identifiers
  tz <- getArchiveOption("tz", "");
  # Example: 2011-03-10GMT
  datestamp <- format(Sys.time(), "%Y-%m-%d", tz=tz);
  datestamp <- sprintf("%s%s", datestamp, tz);

  # Generate archive path
  rootPath <- getArchiveRootPath();
  paths <- list(rootPath, dirs, datestamp);
  paths <- paths[sapply(paths, FUN=length) > 0];
  path <- do.call(file.path, args=paths);
  path <- Arguments$getWritablePath(path);

  path;
}) # getArchivePath()



############################################################################
# HISTORY:
# 2011-03-09
# o Added getArchivePath().
# o Created.
############################################################################
